provider/mesos/config_test.go

```diff
@@ -0,0 +1,310 @@
+				statuses(
+					status(
+						setState("TASK_RUNNING"),
+						setHealthy(true))),
+				setLabels(label.TraefikEnable, "true",
+					label.TraefikPortIndex, "0"),
+				discovery(setDiscoveryPort("TCP", 80, "WEB")),
+			),
+			expected:         true, // traefik.portIndex and discoveryPorts correspond
+			exposedByDefault: true,
+		}, {
+			mesosTask: task(
+				statuses(
+					status(
+						setState("TASK_RUNNING"),
+						setHealthy(true))),
+				setLabels(label.TraefikEnable, "true",
+					label.TraefikPort, "TRAEFIK"),
+				discovery(setDiscoveryPort("TCP", 80, "WEB")),
+			),
+			expected:         false, // traefik.port is not an integer
+			exposedByDefault: true,
+		}, {
+			mesosTask: task(
+				statuses(
+					status(
+						setState("TASK_RUNNING"),
+						setHealthy(true))),
+				setLabels(label.TraefikEnable, "true",
+					label.TraefikPort, "443"),
+				discovery(setDiscoveryPort("TCP", 80, "WEB")),
+			),
+			expected:         false, // traefik.port is not the same as discovery.port
+			exposedByDefault: true,
+		}, {
+			mesosTask: task(
+				statuses(
+					status(
+						setState("TASK_RUNNING"),
+						setHealthy(true))),
+				setLabels(label.TraefikEnable, "true",
+					label.TraefikPort, "80"),
+				discovery(setDiscoveryPort("TCP", 80, "WEB")),
+			),
+			expected:         true, // traefik.port is the same as discovery.port
+			exposedByDefault: true,
+		}, {
+			mesosTask: task(
+				statuses(
+					status(
+						setState("TASK_RUNNING"))),
+				setLabels(label.TraefikEnable, "true",
+					label.TraefikPort, "80"),
+				discovery(setDiscoveryPort("TCP", 80, "WEB")),
+			),
+			expected:         true, // No healthCheck
+			exposedByDefault: true,
+		}, {
+			mesosTask: task(
+				statuses(
+					status(
+						setState("TASK_RUNNING"),
+						setHealthy(false))),
+				setLabels(label.TraefikEnable, "true",
+					label.TraefikPort, "80"),
+				discovery(setDiscoveryPort("TCP", 80, "WEB")),
+			),
+			expected:         false, // HealthCheck at false
+			exposedByDefault: true,
+		},
+	}
+
+	for index, test := range testCases {
+		t.Run(strconv.Itoa(index), func(t *testing.T) {
+			t.Parallel()
+
+			actual := taskFilter(test.mesosTask, test.exposedByDefault)
+			if actual != test.expected {
+				t.Logf("Statuses : %v", test.mesosTask.Statuses)
+				t.Logf("Label : %v", test.mesosTask.Labels)
+				t.Logf("DiscoveryInfo : %v", test.mesosTask.DiscoveryInfo)
+				t.Fatalf("expected %v, got %v", test.expected, actual)
+			}
+		})
+	}
+}
+
+func TestTaskRecords(t *testing.T) {
+	var task = state.Task{
+		SlaveID: "s_id",
+		State:   "TASK_RUNNING",
+	}
+	var framework = state.Framework{
+		Tasks: []state.Task{task},
+	}
+	var slave = state.Slave{
+		ID:       "s_id",
+		Hostname: "127.0.0.1",
+	}
+	var taskState = state.State{
+		Slaves:     []state.Slave{slave},
+		Frameworks: []state.Framework{framework},
+	}
+
+	var p = taskRecords(taskState)
+	if len(p) == 0 {
+		t.Fatal("taskRecord should return at least one task")
```

Comment:
Can you add the word `Function` at the beginning,? Thus the falal log begins with an upper case.